src/compiler/parser.ts

```diff
@@ -4556,6 +4573,101 @@ module ts {
+                // ImportDeclaration:
+                node.importClause = parseImportClause();
+            }
+            node.moduleSpecifier = parseModuleSpecifier();
+            parseSemicolon();
+            return finishNode(node);
+        }
+
+        function parseModuleSpecifier(): StringLiteralExpression {
+            // ModuleSpecifier:
+            //  StringLiteral
+            if (token === SyntaxKind.StringLiteral) {
+                // Ensure the string being required is in our 'identifier' table.  This will ensure 
+                // that features like 'find refs' will look inside this file when search for its name.
+                var moduleSpecifier = <StringLiteralExpression>parseLiteralNode(/*internName*/ true);
+                return moduleSpecifier;
+            }
+
+            parseErrorAtCurrentToken(Diagnostics.String_literal_expected);
+        }
+
+        function parseImportClause(): ImportClause {
+            //ImportClause:
+            //  ImportedDefaultBinding from
+            //  NameSpaceImport from
+            //  NamedImports from
+            //  ImportedDefaultBinding, NameSpaceImport from
+            //  ImportedDefaultBinding, NamedImports from
+
+            var importClause = <ImportClause>createNode(SyntaxKind.ImportClause);
+            if (isIdentifier()) {
+                // ImportedDefaultBinding:
+                //  ImportedBinding
+                importClause.defaultBinding = parseIdentifier();
+            }
+
+            // If there was no default import or if there is comma token after default import 
+            // parse namespace or named imports
+            if (!importClause.defaultBinding ||
+                parseOptional(SyntaxKind.CommaToken)) {
+                importClause.bindings = token === SyntaxKind.AsteriskToken ? parseNamespaceImport() : parseNamedImports();
+            }
+
+            parseExpected(SyntaxKind.FromKeyword);
+            return finishNode(importClause);
+        }
+
+        function parseNamespaceImport(): NamespaceImport {
+            // NameSpaceImport:
+            //  * as ImportedBinding
+            var namespaceImport = <NamespaceImport>createNode(SyntaxKind.NamespaceImport);
+            parseExpected(SyntaxKind.AsteriskToken);
+            parseExpected(SyntaxKind.AsKeyword);
+            namespaceImport.name = parseIdentifier();
+            return finishNode(namespaceImport);
+        }
+
+        function parseNamedImports(): NamedImports {
+            var namedImports = <NamedImports>createNode(SyntaxKind.NamedImports);
+
+            // NamedImports:
+            //  { }
+            //  { ImportsList }
+            //  { ImportsList, }
+
+            // ImportsList:
+            //  ImportSpecifier
+            //  ImportsList, ImportSpecifier
+            parseExpected(SyntaxKind.OpenBraceToken);
+            namedImports.elements = parseDelimitedList(ParsingContext.ImportSpecifiers, parseImportSpecifier);

```

Comment:
you should only both parsing the delimited list if you actually got the open brace token.